pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -640,6 +633,101 @@ func (lb *ListenerBuilder) buildWaypointInboundHTTPFilters(svc *model.Service, c
 	return filters
 }
 
+// buildInboundNetworkFilters generates a TCP proxy network filter on the inbound path
+func (lb *ListenerBuilder) buildWaypointNetworkFilters(svc *model.Service, fcc inboundChainConfig) []*listener.Filter {
+	var svcHostname host.Name
+	var subsetName string
+
+	statPrefix := fcc.clusterName
+	// If stat name is configured, build the stat prefix from configured pattern.
+	if len(lb.push.Mesh.InboundClusterStatName) != 0 {
+		statPrefix = telemetry.BuildInboundStatPrefix(lb.push.Mesh.InboundClusterStatName, fcc.telemetryMetadata, "", uint32(fcc.port.Port), fcc.port.Name)
+	}
+	tcpProxy := &tcp.TcpProxy{
+		StatPrefix:       statPrefix,
+		ClusterSpecifier: &tcp.TcpProxy_Cluster{Cluster: fcc.clusterName},
+	}
+	var destinationRule *networking.DestinationRule
+	if svc != nil {
+		svcHostname = svc.Hostname
+		vs := getVirtualServiceForWaypoint(lb.node.ConfigNamespace, svc, lb.node.SidecarScope.EgressListeners[0].VirtualServices())
+		routes := getWaypointTCPRoutes(vs, fcc.port.Port)
+		if len(routes) > 0 {
+			// Existing (slightly incorrect, but best we can do) semantics.
+			// We are routing to multiple destinations, but want TCPProxy level configuration which is shared.
+			// We have to just pick one to use as the settings.
+			svcHostname = host.Name(routes[0].Destination.Host)
+		}
+		if len(routes) == 1 {
```

Comment:
I see this conditional twice; does it make sense to extract to a var that describes the semantic meaning here? I'm curious why we do this when we only have 1 route specifically

Reply:
Ill add more comments but this is all about compat with the arbitrary sidecar logic basically.

The real problem is weighted clusters don't have any per-upstream config in tcpproxy, while our configs are in DR which is tied to the upstream